diff -Naur acpi.old/actions/ac.sh acpi/actions/ac.sh
--- acpi.old/actions/ac.sh	1969-12-31 19:00:00.000000000 -0500
+++ acpi/actions/ac.sh	2009-08-15 13:20:51.000000000 -0400
@@ -0,0 +1,14 @@
+#!/bin/bash
+# ac on/off event handler
+
+status=`awk '/^state: / { print $2 }' /proc/acpi/ac_adapter/$2/state`
+case $status in
+        "on-line")
+                /sbin/laptop_mode stop
+                exit 0
+        ;;
+        "off-line")
+                /sbin/laptop_mode start
+                exit 0
+        ;;
+esac
diff -Naur acpi.old/actions/battery.sh acpi/actions/battery.sh
--- acpi.old/actions/battery.sh	1969-12-31 19:00:00.000000000 -0500
+++ acpi/actions/battery.sh	2009-08-15 13:27:17.000000000 -0400
@@ -0,0 +1,35 @@
+#! /bin/bash
+# Automatically disable laptop mode when the battery almost runs out.
+
+BATT_INFO=/proc/acpi/battery/$2/state
+
+if [ -f /proc/sys/vm/laptop_mode ] ; then
+   LM=`cat /proc/sys/vm/laptop_mode`
+   if [ $LM -gt 0 ] ; then
+     if [ -f $BATT_INFO ] ; then
+        # Source the config file only now that we know we need
+        if [ -f /etc/default/laptop-mode ] ; then
+                # Debian
+                . /etc/default/laptop-mode
+        elif [ -f /etc/sysconfig/laptop-mode ] ; then
+                # Others
+                . /etc/sysconfig/laptop-mode
+        elif [ -f /etc/laptop-mode.conf ] ; then
+                . /etc/laptop-mode.conf
+        fi
+        MINIMUM_BATTERY_MINUTES=${MINIMUM_BATTERY_MINUTES:-'10'}
+
+        ACTION="`cat $BATT_INFO | grep charging | cut -c 26-`"
+        if [ "$ACTION" = "discharging" ] ; then
+           PRESENT_RATE=`cat $BATT_INFO | grep "present rate:" | sed  "s/.* \([0-9][0-9]* \).*/\1/" `
+           REMAINING=`cat $BATT_INFO | grep "remaining capacity:" | sed  "s/.* \([0-9][0-9]* \).*/\1/" `
+        fi
+	if [ $(($REMAINING * 60 / $PRESENT_RATE)) -lt $MINIMUM_BATTERY_MINUTES ] ; then
+           /sbin/laptop_mode stop
+        fi
+     else
+       logger -p daemon.warning "You are using laptop mode and your battery interface $BATT_INFO is missing. This may lead to loss of data when the battery runs out. Check kernel ACPI support and /proc/acpi/battery folder, and edit /etc/acpi/battery.sh to set BATT_INFO to the correct path."
+     fi
+   fi
+fi
+
diff -Naur acpi.old/events/ac_adapter acpi/events/ac_adapter
--- acpi.old/events/ac_adapter	1969-12-31 19:00:00.000000000 -0500
+++ acpi/events/ac_adapter	2009-08-15 13:18:42.000000000 -0400
@@ -0,0 +1,2 @@
+event=ac_adapter
+action=/etc/acpi/actions/ac.sh %e
diff -Naur acpi.old/events/battery acpi/events/battery
--- acpi.old/events/battery	1969-12-31 19:00:00.000000000 -0500
+++ acpi/events/battery	2009-08-15 13:19:38.000000000 -0400
@@ -0,0 +1,2 @@
+event=battery.*
+action=/etc/acpi/actions/battery.sh %e
